configuration/src/secrets.rs

```diff
@@ -110,18 +115,89 @@ impl AgentSecrets {
 #[cfg(test)]
 mod test {
     use super::*;
-    const SECRETS_JSON_PATH: &str = "../fixtures/test_secrets.json";
-    const SECRETS_ENV_PATH: &str = "../fixtures/env.test";
+    use crate::ethereum::Connection;
+    use nomad_test::test_utils;
+
+    #[test]
+    #[serial_test::serial]
+    fn it_builds_from_env_mixed() {
+        test_utils::run_test_with_env_sync("../fixtures/env.test-signer-mixed", move || {
+            let networks = &crate::get_builtin("test").unwrap().networks;
+            let secrets =
+                AgentSecrets::from_env(networks).expect("Failed to load secrets from env");
+
+            assert_eq!(
+                *secrets.transaction_signers.get("moonbeam").unwrap(),
+                SignerConf::Aws {
+                    id: "moonbeam_id".into(),
+                    region: "moonbeam_region".into(),
+                }
+            );
+            assert_eq!(
+                *secrets.transaction_signers.get("ethereum").unwrap(),
+                SignerConf::HexKey(
+                    "0x1111111111111111111111111111111111111111111111111111111111111111"
+                        .parse()
+                        .unwrap()
+                )
+            );
+            assert_eq!(
+                *secrets.transaction_signers.get("evmos").unwrap(),
+                SignerConf::Aws {
+                    id: "default_id".into(),
+                    region: "default_region".into(),
+                }
+            );
+            assert_eq!(
+                *secrets.rpcs.get("moonbeam").unwrap(),
+                ChainConf::Ethereum(Connection::Http("https://rpc.api.moonbeam.network".into()))
+            );
+            assert_eq!(
+                *secrets.rpcs.get("ethereum").unwrap(),
+                ChainConf::Ethereum(Connection::Http(
+                    "https://main-light.eth.linkpool.io/".into()
+                ))
+            );
+            assert_eq!(
+                *secrets.rpcs.get("evmos").unwrap(),
+                ChainConf::Ethereum(Connection::Http("https://eth.bd.evmos.org:8545".into()))
+            );
+        });
+    }
+
+    #[test]
+    #[serial_test::serial]
+    fn it_builds_from_env_default() {
```

Comment:
Could we add a default test case to `kathy/src/settings.rs`? We've been using kathy settings as place for testing full e2e configuration of agent from env to agent settings block. We'd just need to essentially copy [this test](https://github.com/nomad-xyz/rust/blob/main/agents/kathy/src/settings.rs#L17-L48) except piping in `test-signer-default` test fixture.

Reply:
Sounds good. Might have some q's